### PR TITLE
feat: add shellcheck make target

### PR DESCRIPTION
**Pull request by vprashar2929**
This PR adds `shellcheck` target which installs shellcheck tool locally. `make lint` will also run shellcheck

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -52,9 +58,10 @@ pkg/rest/zz_docs: $(wildcard pkg/rest/*.go pkg/korrel8r/*.go) $(SWAG)
 	$(SWAG) fmt pkg/rest
 	@touch $@
 
-lint: $(VERSION_TXT) $(GOLANGCI_LINT) $(SHFMT) ## Run the linter to find and fix code style problems.
+lint: $(VERSION_TXT) $(GOLANGCI_LINT) $(SHFMT) shellcheck ## Run the linter to find and fix code style problems.
```
Comment (alanconway) [inline Makefile:61]: Use Make dependenciees to download shellcheck:
```suggestion
SHELLCHECK=$(LOCALBIN)/shellcheck
lint: $(VERSION_TXT) $(GOLANGCI_LINT) $(SHFMT) $(SHELLCHECK) ## Run the linter to find and fix code style problems.
```

$(SHELLCHECK):
        hack/download_shellcheck.sh $(LOCALBIN)/shellcheck

File: Makefile [hunk 2]
```diff
@@ -132,5 +139,25 @@ tag-release:
 	hack/tag-release.sh $(VERSION)
 	$(IMGTOOL) push -q "$(IMAGE)" "$(IMG):latest"
 
+.PHONY: shellcheck
```
Comment (alanconway) [inline Makefile:142]: Move to shell script hack/download_shellcheck.sh, no need to check for existence this make rule will only be invoked if shellcheck is missing

**Review by alanconway — CHANGES_REQUESTED** [review]
Minor suggestion, otherwise great.

**Comment by vprashar2929** [comment]
cc: @alanconway if it looks good can we merge this?

**Pull request merged**